### PR TITLE
ci: use stable branch instead tags

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Use a branch instead of a tag to avoid the RSS issues with tags/releases in GitHub

### CODE REVIEW

**Review by SylvainJuge — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Comment by v1v** [comment]
I'll test this PR with https://github.com/elastic/apm-agent-java/pull/1771/commits/2310c66d4111cc561283b8819e5bcb8dd1eef0ba then I'll revert it

**Comment by v1v** [comment]
### Tests

When branch does not exist

image

**Comment by SylvainJuge** [comment]
As I just mentioned in the related robots issue, we should also make sure that this `stable` branch do not get built in the main multi-branch pipeline.

File: .ci/jobs/apm-agent-java-mbp.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@
             discover-pr-forks-trust: permission
             discover-pr-origin: merge-current
             discover-tags: true
+            head-filter-regex: '^(?!stable).*$'
```
Comment (v1v) [inline .ci/jobs/apm-agent-java-mbp.yml:16]: Exclude the branch to be built in the CI.

**Review by cachedout — APPROVED** [review]

**Pull request merged**